src/Goteo/Payment/Method/AbstractPaymentMethod.php

```diff
@@ -22,39 +23,49 @@
 use Omnipay\Common\GatewayFactory;
 use Omnipay\Common\GatewayInterface;
 use Omnipay\Common\Message\ResponseInterface;
+use Omnipay\GuzzleClient;
 use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
 
 /**
- * Helper class with common of the interface methods implemented in a simple way
+ * Helper class with some common interface methods
  *
  * Payments using this implementation use the Omnipay library:
  * http://omnipay.thephpleague.com/
  */
 abstract class AbstractPaymentMethod implements PaymentMethodInterface
 {
     protected ?GatewayInterface $gateway = null;
-    protected $invest;
-    protected $request;
-    protected $user;
+    protected ?Invest $invest;
+    protected ?Request $request;
+    protected ?User $user;
 
     /**
      * @throws PaymentException
+     * @throws Config\ConfigException
      */
     public function __construct(User $user = null) {
         $this->user = $user;
         $this->initGateway();
     }
 
+    /**
+     * @throws PaymentException
+     * @throws Config\ConfigException
+     */
     private function initGateway()
     {
         $factory = new GatewayFactory();
-        $this->gateway = $factory->create($this->getGatewayName());
+        $this->gateway = $factory->create(
+            $this->getGatewayName(),
+//            new GuzzleClient() // @TODO Move GuzzleClient from goteo-private into goteo
```

Comment:
It seems this is where the HttpClient is set. However, if we let the GatewayFactory with a null HttpClient, it's unclear what happens afterwards...